src/services_models/reserves.jl

```diff
@@ -81,9 +96,113 @@ function service_requirement_constraint!(
     return
 end
 
+function service_requirement_constraint!(
+    psi_container::PSIContainer,
+    service::SR,
+    model::ServiceModel{SR, StepwiseCostReserve},
+) where {SR <: PSY.Reserve}
+    parameters = model_has_parameters(psi_container)
+    use_forecast_data = model_uses_forecasts(psi_container)
+    initial_time = model_initial_time(psi_container)
+    @debug initial_time
+    time_steps = model_time_steps(psi_container)
+    name = PSY.get_name(service)
+    constraint = get_constraint(psi_container, constraint_name(REQUIREMENT, SR))
+    reserve_variable = get_variable(psi_container, variable_name(name, SR))
+    requirement_variable =
+        get_variable(psi_container, variable_name(SERVICE_REQUIREMENT, SR))
+
+    for t in time_steps
+        constraint[name, t] = JuMP.@constraint(
+            psi_container.JuMPmodel,
+            sum(reserve_variable[:, t]) >= requirement_variable[name, t]
+        )
+    end
+
+    return
+end
+
+function cost_function(
+    psi_container::PSIContainer,
+    service::SR,
+    ::Type{StepwiseCostReserve},
+) where {SR <: PSY.Reserve}
+
+    use_forecast_data = model_uses_forecasts(psi_container)
+    initial_time = model_initial_time(psi_container)
+    @debug initial_time
+    time_steps = model_time_steps(psi_container)
+
+    if use_forecast_data
+        ts_vector = _convert_to_variablecost(PSY.get_data(PSY.get_forecast(
+            PSY.PiecewiseFunction,
+            service,
+            initial_time,
+            "get_variable",
+            length(time_steps),
+        )))
+
+    else
+        ts_vector = repeat(PSY.get_variable(PSY.get_op_cost(service)), time_steps[end])
+    end
+
+    resolution = model_resolution(psi_container)
+    dt = Dates.value(Dates.Minute(resolution)) / 60
```

Comment:
There is a conversion function in IS to do this conversion. Also. please don't divide by integers, there is a const in `definitions.jl` for `MINUTES_IN_HOUR`